src/main/java/ch/wisv/areafiftylan/users/service/UserServiceImpl.java

```diff
@@ -70,8 +69,9 @@ public User getUserById(long id) {
     }
 
     @Override
-    public Optional<User> getUserByUsername(String username) {
-        return userRepository.findOneByUsernameIgnoreCase(username);
+    public User getUserByUsername(String username) {
+        return userRepository.findOneByUsernameIgnoreCase(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User '" + username + "' not found"));
```

Comment:
Very nice alteration, I was wondering how you were able to remove the throw statement in TeamServiceImpl.